process/block/sovereignblock.go

```diff
@@ -0,0 +1,451 @@
+	if check.IfNil(initialHdr) {
+		return nil, nil, process.ErrNilBlockHeader
+	}
+	commonHdr, ok := initialHdr.(*block.HeaderWithValidatorStats)
+	if !ok {
+		return nil, nil, process.ErrWrongTypeAssertion
+	}
+
+	s.processStatusHandler.SetBusy("sovereignBlockProcessor.CreateBlock")
+	defer s.processStatusHandler.SetIdle()
+
+	for _, accounts := range s.accountsDB {
+		if accounts.JournalLen() != 0 {
+			log.Error("sovereignBlockProcessor.CreateBlock first entry", "stack", accounts.GetStackDebugFirstEntry())
+			return nil, nil, process.ErrAccountStateDirty
+		}
+	}
+
+	err := s.createBlockStarted()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	s.blockChainHook.SetCurrentHeader(commonHdr)
+	startTime := time.Now()
+	mbsFromMe := s.txCoordinator.CreateMbsAndProcessTransactionsFromMe(haveTime, commonHdr.GetPrevRandSeed())
+	elapsedTime := time.Since(startTime)
+	log.Debug("elapsed time to create mbs from me", "time", elapsedTime)
+
+	return initialHdr, &block.Body{MiniBlocks: mbsFromMe}, nil
+}
+
+// ProcessBlock actually process the selected transaction and will create the final block body
+func (s *sovereignBlockProcessor) ProcessBlock(
+	headerHandler data.HeaderHandler,
+	bodyHandler data.BodyHandler,
+	haveTime func() time.Duration,
+) error {
+	if haveTime == nil {
+		return process.ErrNilHaveTimeHandler
+	}
+
+	s.processStatusHandler.SetBusy("shardProcessor.ProcessBlock")
+	defer s.processStatusHandler.SetIdle()
+
+	err := s.checkBlockValidity(headerHandler, bodyHandler)
+	if err != nil {
+		if err == process.ErrBlockHashDoesNotMatch {
+			go s.requestHandler.RequestShardHeader(headerHandler.GetShardID(), headerHandler.GetPrevHash())
+		}
+
+		return err
+	}
+
+	blockBody, ok := bodyHandler.(*block.Body)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	err = s.createBlockStarted()
+	if err != nil {
+		return err
+	}
+
+	s.txCoordinator.RequestBlockTransactions(blockBody)
+	err = s.txCoordinator.IsDataPreparedForProcessing(haveTime)
+	if err != nil {
+		return err
+	}
+
+	for _, accounts := range s.accountsDB {
+		if accounts.JournalLen() != 0 {
+			log.Error("metaProcessor.CreateBlock first entry", "stack", accounts.GetStackDebugFirstEntry())
+			return process.ErrAccountStateDirty
+		}
+	}
+
+	defer func() {
+		if err != nil {
+			s.RevertCurrentBlock()
+		}
+	}()
+
+	startTime := time.Now()
+	err = s.txCoordinator.ProcessBlockTransaction(headerHandler, blockBody, haveTime)
+	elapsedTime := time.Since(startTime)
+	log.Debug("elapsed time to process block transaction",
+		"time [s]", elapsedTime,
+	)
+	if err != nil {
+		return err
+	}
+
+	s.prepareBlockHeaderInternalMapForValidatorProcessor()
+	_, err = s.validatorStatisticsProcessor.UpdatePeerState(headerHandler, makeCommonHeaderHandlerHashMap(s.hdrsForCurrBlock.getHdrHashMap()))
+	if err != nil {
+		return err
+	}
+
+	finalBody, err := s.applyBodyToHeader(headerHandler, blockBody)
+	if err != nil {
+		return err
+	}
+	blockBody = finalBody
```

Comment:
This will work as expected? blockBody is not returned. Also I think we can use directly bolckBody in L166 and remove finalBody at all

Reply:
changed with SET